File: ecommerce-frontend/src/services/categoryService.js
import API from "./api";

export const getAllCategories = () => API.get("/categories").then((r) => r.data);
export const createCategory = (data) => API.post("/categories", data).then((r) => r.data);
export const updateCategory = (id, data) =>
  API.put(`/categories/${id}`, data).then((r) => r.data);
export const deleteCategory = (id) => API.delete(`/categories/${id}`).then((r) => r.data);

File: ecomerceBacked/src/main/java/com/ecomerce/ecomerceBacked/service/CategoryService.java
package com.ecomerce.ecomerceBacked.service;

import com.ecomerce.ecomerceBacked.model.Category;
import com.ecomerce.ecomerceBacked.repository.CategoryRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CategoryService {
    private final CategoryRepository categoryRepository;

    public CategoryService(CategoryRepository categoryRepository) {
        this.categoryRepository = categoryRepository;
    }
    public List<Category> getAllCategories() {
        return categoryRepository.findAll();
    }
    public Category createCategory(Category category) {
        return categoryRepository.save(category);
    }
    public Category updateCategory(Long id, Category category) {
        category.setId(id);
        return categoryRepository.save(category);
    }
    public void deleteCategory(Long id) {
        categoryRepository.deleteById(id);
    }
}
